### ISSUE TITLE
Calling Quick Compile & Debug multiple times fails when launching SWF embedded in HTML

### ISSUE THREAD

**Issue by jcrawford** [BowlerHatLLC/vscode-nextgenas#277]
When using the Quick Compile & Debug feature I get errors on subsequent runs.
Steps to reproduce are below.

1. Open Code
2. Command+Shift+P select Quick Compile & Debug
3. Project Compiles and Browser Opens
4. Close the Browser window/tab
5. Command+Shift+P select Quick Compile & Debug
6. See the following error message

<img width="465" alt="screen shot 2018-09-17 at 12 37 54 pm" src="https://user-images.githubusercontent.com/875341/45642906-acf6b200-ba87-11e8-8f3f-be0dce878c62.png">

7. Close Code & Reopen
8. Command+Shift+P Choose Quick Compile & Debug
9. Project Compiles and Browser Opens
10. Loop back to step 4

This seems to happen every time I follow the above steps.  It appears that it is a problem with the Quick Compile & Debug since when I click the green arrow in the debug dialog a browser window opens every time.

I have the following set in my tasks and launch files.

`tasks.json`

```
{
    "version": "2.0.0",
    "tasks": [
        {
            "identifier": "build-debug",
            "type": "actionscript",
            "debug": false,
            "problemMatcher": [
                "$nextgenas_nomatch"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            }
        }
    ]
}
```

`launch.json`

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "swf",
            "request": "launch",
            "name": "Launch SWF",
            "program": "bin/manufacturer.html",
            "runtimeExecutable": "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"
        },
        {
            "type": "swf",
            "request": "attach",
            "name": "Attach SWF"
        }
    ]
}
```

**Comment by joshtynjala** [comment]
Thanks! I'm guessing that the green arrow automatically terminates the existing debug session, and Quick Compile & Debug does not yet. Either that, or you need to press stop before VSCode even allows you to start a new session with the green arrow again, which would pretty much have the same effect.

**Comment by jcrawford** [comment]
I am not sure if that is the issue or not honestly.  I clicked Debug > Stop Debugging and it didn't say whether it stopped or not.  I then tried the green play button after hitting this issue and it as well showed the same error dialog.  Now when I restart Code it still refuses to debug.

**Comment by joshtynjala** [comment]
Could it be that the browser is still open at your previous debugging session?

**Comment by jcrawford** [comment]
No I close the tab every time I go through those steps but the browser window is open so if Chrome retained anything that could be the cause..  I doubt it though.

**Comment by jcrawford** [comment]
I think I see what is happening, you are correct in the fact that your plugin is not stopping the debugging session, it is also not closing down java after the compile.  However our SWF takes several minutes to load everything and I am guessing that Code is waiting for that process.  I say this because after a few minutes when the flex component is loaded in the browser I can close the tab and compile again.

**Comment by joshtynjala** [comment]
Yeah, that could be what's happening. You could give Firefox or Safari a try instead, to see if it works better. Just make sure that you have the debug version of the Flash Player plugin installed, of course, or VSCode won't be able to connect.

The NPAPI plugin architecture used by Firefox/Safari is different than the PPAPI plugins in Chrome, so any issues that might affect Chrome may not affect other browsers.

Alternatively, you could launch your SWF in the standalone Flash Player instead of a browser. It's probably faster too.

**Comment by jcrawford** [comment]
Good Idea I might have to look into launching it into the standalone flash player did not think about that.  Would it work the same way as the launcher for chrome except I specify the path to the player?

**Comment by joshtynjala** [comment]
Pretty much, yeah. You might not even need to specify the path to Flash Player, if SWF files are already associated with it at the operating system level.

**Comment by joshtynjala** [comment]
Relevant documentation:

https://github.com/BowlerHatLLC/vscode-nextgenas/wiki/Debug-a-SWF-in-the-standalone-Adobe-Flash-Player-with-Visual-Studio-Code

**Comment by jcrawford** [comment]
Actually this may not work, I need access to dev tools to see ajax requests that are being made to our API layer, I do not see the ability for that in flash player.

**Comment by joshtynjala** [comment]
When running Quick Compile & Debug, I'm now forcing the debugger to stop before it compiles and starts a new session. This should give you the expected behavior. It's already improving my workflow with an AIR app that I don't always remember to close before hitting Ctrl+Enter.

This change will be included in the v0.14 update, which I plan to release within the next week.

**Comment by jcrawford** [comment]
Josh,

That is awesome to hear thank you very much it will make working with this plugin so much better!
